geomesa-kafka-bundle/geomesa-kafka-processors/pom.xml

```diff
@@ -15,13 +15,104 @@
     <description>GeoMesa Kafka processor implementations</description>
 
     <dependencies>
+
+        <dependency>
+            <groupId>org.geomesa.nifi</groupId>
+            <artifactId>geomesa-datastore-services-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
         <dependency>
             <groupId>org.geomesa.nifi</groupId>
             <artifactId>geomesa-datastore-processors</artifactId>
         </dependency>
+
         <dependency>
             <groupId>org.locationtech.geomesa</groupId>
             <artifactId>geomesa-kafka-datastore_${scala.binary.version}</artifactId>
+            <exclusions>
+                <!-- exclude dependencies provided by datastore-services-api-nar -->
+                <exclusion>
```

Comment:
The part of me that's too quick wants to use an <exclusion>*:*</exclusion>.  I think those are deprecated, but meh.

Also, we still need some of the dependencies.... right?  That's too quick, right?

Reply:
yeah, we still need some of them. this was a PITA... i manually checked them for overlap by `unzip -l` the nars and comparing them against the dependency tree for the geomesa-datastore-services-api module. I believe having duplicates would not really matter as long as they are exactly the same (as that was the current state of things before this PR), but it cuts down on the nar size a lot.

Reply:
(Non-blocking question.)  Whew... is there a good way to automate this going forward? 

I suppose I'd be worried that we'll miss a dependency later.

Reply:
i did not find a good way, and indeed did not get it down to zero overlapping dependencies. this is quite annoying, which is probably why i thought the nifi nar plugin would handle it (since it knows all the dependencies)

Reply:
i think in most cases, it would just mean a larger nar, but wouldn't break functionality since the versions of the jars would be the same due to maven resolution. but that might not always be true, i.e. if the jar is trying to load something from geotools that is only available in the parent nar.

Reply:
Hmm.... I wonder if the nifi nar plugin should be improved to do what we want it to do?

We can be the change we want to see in the world.